### ISSUE TITLE
Some questions about nebulet

### ISSUE THREAD

**Issue by Frederik-Baetens** [nebulet/nebulet#74]
Is nebulet still in active development? I saw this tweet: https://twitter.com/willscott/status/1121607193873145856 so I assume it is/will be?

Do you think nebulet can implement WASI?

And how much memory overhead does nebulet produce? Could wasm eventually be run on very resource constrained devices? 100-500kb flash?

**Comment by chpio** [comment]
> Is nebulet still in active development? I saw this tweet: https://twitter.com/willscott/status/1121607193873145856 so I assume it is/will be?

ohh, cool

> Do you think nebulet can implement WASI?

Don't think so, at least not without thread support.

**Comment by edomaur** [comment]
I'm also wondering about the future of this project...

**Comment by lachlansneff** [comment]
Hi everyone,

It's been a while since I've looked at this repository.

Nebulet is not in active development, and hasn't been for a bit more than a year. There are a few reasons for this, but the main one is that I felt it had fulfilled its purpose: to demonstrate that microkernels that run managed code in kernel-mode are viable (before spectre/meltdown ruined things), at least to some extent. (Additionally, it helped me get internships, jobs, etc)

As it stands right now, I don't have the time or the motivation to work on Nebulet. My interests have moved on to other things, primarily the space industry (please message me [email in profile] if anyone who reads this works in the space industry, looking for internships/co-ops).

If someone would like to take on the Nebulet banner and continue to work on it, I'd be happy to pass it on. Otherwise, it'll probably sit here for the foreseeable future and will likely be archived at some point.

\- Lachlan Sneff

**Comment by Frederik-Baetens** [comment]
Cool, thanks a lot for working on this, i wish you well in your further life!

For those that were hoping that this could lead to wasm on resource constrained devices, nebulet is probably also made obsolete by [wasm-micro-runtime](https://github.com/bytecodealliance/wasm-micro-runtime)

**Comment by chpio** [comment]
@Frederik-Baetens yeah, but it's still just an interpreter, compiling a wasm ahead-of-time into a fully integrated unikernel, like proposed [here](https://github.com/nebulet/nebulet/issues/72#issuecomment-423304498) would be really cool.

edit: maybe it does support ahead-of-time compilation... https://github.com/bytecodealliance/wasm-micro-runtime/blob/master/doc/build_wasm_app.md#compile-wasm-to-aot-module
